app/models/katello/glue/pulp/repo.rb

```diff
@@ -47,16 +47,25 @@ def self.needs_distributor_updates(repos, smart_proxy)
 
         def self.build_override_config(options)
           config = {}
-          if options[:filters].present? && (options[:solve_dependencies] || options[:resolve_dependencies])
+          if options[:force_dep_solve] ||
+              (options[:filters].present? && (options[:solve_dependencies] || options[:resolve_dependencies]))
             if Setting[:dependency_solving_algorithm] == 'greedy'
               config[:recursive] = true
             else
               config[:recursive_conservative] = true
             end
+            config[:additional_repos] = options[:repository_mapping] unless options[:repository_mapping].blank?
           end
-
           config
         end
+
+        def self.generate_yum_dependent_repositories(cloned_repos = {})
+          ret = {}
+          cloned_repos.each do |source, dest|
+            ret[source.first.pulp_id] = dest.pulp_id if source.first.yum?
+          end
+          ret
+        end
```

Comment:
does this need to go in the glue layer?  Also, can it refer to repositories and NOT pulp ids?  Leave it to the service layer to figure out what is needed.

Reply:
if this returned a "RepositoryMapper" object, you could also convert it to a pulp_id -> pulp_id hash if that is useful in a service layer.

Reply:
I need a way to serialize. I originally had {soure_repo:destination_repo} full blown objects. If you want I can do {source_repo_id:destination_repo_id} and then hav ethe service layer convert that